frontend/src/components/widgets/options/ToolOptions.vue

```diff
@@ -134,9 +149,9 @@ export default defineComponent({
 					props: {},
 				},
 			],
-			Shape: [{ kind: "NumberInput", callback: this.setShapeOptions, props: { value: 6, min: 3, isInteger: true, label: "Sides" } }],
-			Line: [{ kind: "NumberInput", callback: this.setLineOptions, props: { value: 5, min: 1, isInteger: true, unit: " px", label: "Weight" } }],
-			Pen: [{ kind: "NumberInput", callback: this.setPenOptions, props: { value: 5, min: 1, isInteger: true, unit: " px", label: "Weight" } }],
+			Shape: [{ kind: "NumberInput", path: ["shape_type", "Polygon", "vertices"], props: { min: 3, isInteger: true, label: "Sides" } }],
```

Comment:
These "paths" correspond to options within variants of the `ToolOptions` enum in Rust. `Polygon` is a variant of an enum (`ShapeType`, while the other names are fields in structs. This will also become much more sensible when all the option definitions and layouts are in the backend.